unit_tests/test_configuration.py

```diff
@@ -34,6 +36,46 @@ def test_bootstrap(self):
             FileContains(matcher=Contains("<numExecutors>1</numExecutors>")))
         self.assertEqual({8080}, self.fakes.juju.ports["TCP"])
 
+    def test_set_prefix(self):
+        test_file = tempfile.NamedTemporaryFile()
```

Comment:
Yeah, I should have mentioned that the case were it hangs around is if the test method fails. In that case the close() call never gets executed (an exception is raised before).